packages/vanilla/rollup.config.mjs

```diff
@@ -1,17 +1,36 @@
 import { babel } from '@rollup/plugin-babel'
-import ts from 'rollup-plugin-typescript2'
 import pkg from './package.json' assert { type: 'json' }
+import typescript from '@rollup/plugin-typescript'
 
-const config = {
-  input: 'src/index.ts',
-  output: [
-    { file: pkg.main, format: 'cjs' },
-    { file: pkg.module, format: 'esm' },
-  ],
-  plugins: [
-    ts({ tsconfigOverride: { exclude: ['**/*-test.ts'] } }),
-    babel({ babelHelpers: 'inline' }),
-  ],
-}
+const config = [
+  // config for CommonJS
+  {
+    input: 'src/index.ts',
+    output: { file: pkg.main, format: 'cjs', generatedCode: 'es5' },
+    plugins: [
+      // run TS first
+      typescript({
+        tsconfig: './tsconfig.cjs.json', // this TS config checks types only, Babel compiles in the next plugin
+        exclude: ['**/*-test.ts'],
+      }),
+      // compile with Babel (see babel.config.js)
+      babel({ babelHelpers: 'inline', include: ['.ts'] }),
+    ],
+  },
+  // config for ES Modules
+  {
+    input: 'src/index.ts',
+    output: { file: pkg.module, format: 'esm', generatedCode: 'es2015' },
```

Comment:
If the environment can load ES Modules, then it also supports things like Object.assign, arrow functions, etc. and hence doesn't need lowered. The output in `./dist` is nearly identical to the original TS.